esp-config/src/bin/esp-config/tui.rs

```diff
@@ -0,0 +1,758 @@
+                                self.handle_error(set_res);
+                            }
+                            self.showing_selection_popup = false;
+                        }
+                        _ => (),
+                    }
+                } else if self.show_error_message {
+                    match key.code {
+                        KeyCode::Enter if key.kind == KeyEventKind::Press => {
+                            self.show_error_message = false;
+                        }
+                        _ => (),
+                    }
+                } else if key.kind == KeyEventKind::Press {
+                    use KeyCode::*;
+
+                    if self.confirm_quit {
+                        match key.code {
+                            Char('y') | Char('Y') => return Ok(None),
+                            _ => self.confirm_quit = false,
+                        }
+                        continue;
+                    }
+
+                    match key.code {
+                        Char('q') => self.confirm_quit = true,
+                        Char('s') | Char('S') => return Ok(Some(self.repository.configs.clone())),
+                        Esc => {
+                            if self.state.len() == 1 {
+                                self.confirm_quit = true;
+                            } else {
+                                self.repository.up();
+                                self.exit_menu();
+                            }
+                        }
+                        Char('h') | Left => {
+                            self.repository.up();
+                            self.exit_menu();
+                        }
+                        Char('l') | Char(' ') | Right | Enter => {
+                            let selected = self.selected();
+                            if self.repository.is_option(selected) {
+                                let current = self.repository.current_level()[selected].value();
+                                let constraint =
+                                    self.repository.current_level()[selected].constraint();
+
+                                match current {
+                                    Value::Bool(value) => {
+                                        let set_res = self
+                                            .repository
+                                            .set_current(selected, Value::Bool(!value));
+                                        self.handle_error(set_res);
+                                    }
+                                    Value::Integer(value) => {
+                                        let display_value = match self.repository.current_level()
+                                            [selected]
+                                            .display_hint()
+                                        {
+                                            DisplayHint::None => format!("{value}"),
```

Comment:
This is not the first time I've seen this logic. And given that the library parses octal numeric literals, it's also not the only place where this isn't complete.